flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionBuilder.java

```diff
@@ -112,13 +114,19 @@ public ResultPartitionBuilder setBufferPoolFactory(
 		return this;
 	}
 
+	public ResultPartitionBuilder isReleasedOnConsumption(boolean releasedOnConsumption) {
+		this.releasedOnConsumption = releasedOnConsumption;
+		return this;
+	}
+
 	public ResultPartition build() {
 		ResultPartitionFactory resultPartitionFactory = new ResultPartitionFactory(
 			partitionManager,
 			ioManager,
 			networkBufferPool,
 			networkBuffersPerChannel,
-			floatingNetworkBuffersPerGate);
+			floatingNetworkBuffersPerGate,
+			true);
```

Comment:
After this value is set true, it seems no way to generate `ResultPartition` atm via builder no matter with `releasedOnConsumption` is set true or false.
Another option is make this value false, and let `releasedOnConsumption` default value is true, then the final tag could  also be changed via `ResultPartitionBuilder#isReleasedOnConsumption(false)`.
I am not sure whether to adjust this issue atm or fix when required future.